src/compiler/crystal/codegen/fun.cr

```diff
@@ -382,7 +382,11 @@ class Crystal::CodeGenVisitor
   def setup_context_fun(mangled_name, target_def, llvm_args_types, llvm_return_type) : Nil
     context.fun = @llvm_mod.functions.add(mangled_name, llvm_args_types, llvm_return_type, target_def.varargs?)
 
-    context.fun.add_attribute LLVM::Attribute::AlwaysInline if target_def.always_inline?
+    unless @debug.none?
+      context.fun.add_attribute LLVM::Attribute::NoInline
```

Comment:
You can't introduce such a drastic change.

We always have some debug information by default (function name, file:line), and we still want LLVM to decide whether it inlines a function or not.

Reply:
This is enabled only when you are in debug mode. When you pass --release flag it will behave as usual. It is not critical though here, so I can remove it.

Reply:
You are right, we should let LLVM optimizer decide if it needs to be inlined or not.
I just removed that code.